### PR TITLE
Add more pre-commit checks

### PR DESCRIPTION
**Pull request by Zethson**
Hey,

finally solves #1563 

I added: 

```
    -   id: trailing-whitespace
    -   id: end-of-file-fixer
    -   id: check-added-large-files
    -   id: check-case-conflict
    -   id: check-toml
    -   id: check-yaml
    -   id: check-merge-conflict
    -   id: detect-private-key
```

They are super super fast, so speed is not a concern.

Signed-off-by: zethson <lukas.heumos@posteo.net>

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -12,3 +12,15 @@ repos:
     hooks:
     -   id: autopep8
         args: ["-i"]
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-added-large-files
+        exclude: docs/data
```
Comment (ivirshup) [inline .pre-commit-config.yaml]: ?

We don't want any large files in there. Hopefully none got through with all the plotting example PRs...

It does look like you might have added one here: `docs/data/pbmc3k_raw.h5ad`

**Review by ivirshup — CHANGES_REQUESTED** [review]
Only thing that needs to change is the added large file. 

Could you try building the docs and checking if anything gets added? The `scanpy.plotting.rst` file has been weird in the past. I think it's at least started being consistent with newer versions of sphinx, but it'd be good to check.

I would like it if the `yaml` and `toml` checks would also do formatting, but up to you on including that.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Zethson) [reply .pre-commit-config.yaml]: Yeah, true.
Think this one origins from me building the docs. I actually thought that this was already there, but apparently not. Hence, I excluded this path.

I will remove the file & remove the exclude.

**Comment by Zethson** [comment]
> Could you try building the docs and checking if anything gets added? The `scanpy.plotting.rst` file has been weird in the past. I think it's at least started being consistent with newer versions of sphinx, but it'd be good to check.

The pbmc data in the docs folder does indeed get downloaded when building the docs.
Do you want me to include this path into the `.gitignore`?

**Comment by ivirshup** [comment]
> Do you want me to include this path into the .gitignore?

:+1:

**Comment by Zethson** [comment]
> > Do you want me to include this path into the .gitignore?
> 
> +1

Done.

**Comment by ivirshup** [comment]
One last thing, could you exclude the test data files from any automatic formatting? I guess it's good to know that new lines at the end of files doesn't matter, but I'd prefer to keep those files exactly as they were written by `spaceranger`/ `zarr`.

**Comment by Zethson** [comment]
@ivirshup 

`exclude: scanpy/tests/_data`

Added for trailing-whitespace and end-of-file-fixer.

**Comment by ivirshup** [comment]
Do revert the changes those made as well please!

**Comment by Zethson** [comment]
@ivirshup Done.

**Comment by ivirshup** [comment]
Great, thanks

**Pull request merged**